libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -536,4 +540,19 @@ void lua_scripts::run(void) {
             }
         }
     }
+
+    // make sure all scripts have been removed
+    while (scripts != nullptr) {
+        remove_script(lua_state, scripts);
+    }
```

Comment:
This makes me a bit itchy.  Looks like an infinite loop right here - obviously not as remove_script must be taking a reference...

Reply:
`scripts` is a linked list of the scripts to be run, `remove_script` takes a pointer to the script we want to dequeue from the list, but it also has a special case to handle when the passed script is the head of the list, which is why this isn't an infinite loop. (The other good news is this is inside the scripting low priority thread still)